astropy/tests/tests/test_runner.py

```diff
@@ -10,6 +10,9 @@
 from astropy.tests.runner import keyword
 
 
+@pytest.mark.filterwarnings(
+    "ignore:The TestRunner.* class:astropy.utils.exceptions.AstropyPendingDeprecationWarning"
```

Comment:
So... even a pending deprecation warning may fail downstream CI if they turn all unhandled warnings into failures. Is v7.1 still appropriate for pending deprecation to go in, or we have to wait till v8.0?

Reply:
Oh, didn't know about that one. Yes, we should use it.

Reply:
Oh but wait, it is a context manager, so I would have to indent almost everything, making the diff harder to review. Not sure if I like that.

Reply:
It already is checking for that, just a different syntax. Warning is a warning.

Reply:
Oh you mean, throw error if the warning doesn't pop up? Hmm. I guess that is better than ignore.